### DISCUSSION TITLE
Wording change and clarification for purchasing Immich

### DISCUSSION THREAD

**Discussion by alextran1502** [immich-app/immich#11313] [Announcements]
Hello everybody,

Alex here! After the last announcement on a wording change in response to our mishap of using the words `licensed` and `unlicensed`, the team and I gave it a long hard thought over the weekend. 

I thought about it when I went to bed, I thought about it in my dream, I thought about it while doing the dishes. Taking into consideration all of your feedback and suggestions, and on behalf of the team, I am happy to show you what we came up with.

- There will be no `license` or `unlicensed`.
- We will change `License Key` to `Product Key`.
- We will show a "Buy Immich" button as the call to action. The button will only be shown if the account age is older than 14 days.
- We will give you options to hide the button for 30 days or indefinitely.

![buttontext](https://github.com/user-attachments/assets/9bd5a296-ca95-45a8-81fb-94b4dab6f05e)

- We will change the price to a whole number `$100` and `$25`. So get your product key now and you can save `$0.01` 😄
- We will add text to make our mission and intentions clear.
- We will refine the text about each option to clarify the content.

![infopanel](https://github.com/user-attachments/assets/fde501aa-e2a6-4999-bba9-2886b5660846)

Last but not least, we will add a fancy `Supporter Badge` to recognize those who purchase the product. If you have any ideas to make it "shinier," let us know!

https://github.com/user-attachments/assets/ff1edb43-1838-45b9-85ed-1944645291f8

Thank you again for your support and feedback. We are working on these changes and will roll out in the next release in a few days.

Have a good week!

Cheers,
Immich Team

**Comment by alepouna** [comment]
The good ending 👍

**Comment by sym0nd0** [comment]
Thanks for listening and acting on our concerns. Very happy to support Immich.

**Comment by othyn** [comment]
This is awesome! Thank you for being so reactive to the communities feedback and coming to a worthwhile compromise. Awesome to see, well done!

**Comment by truppelito** [comment]
I read through the discussion page when the change was announced and this seems to positively address all the concerns I remember being posted there. Thank you!

**Comment by Julian-1-2-3-4-5** [comment]
This is perfect thank you for listening and coming up with a good solution^^

**Comment by afonsofrancof** [comment]
Good change, just think it would be better to call it “Support Immich” instead of “Buy Immich”

**Reply by alextran1502** [reply]
We don't want to give the impression that we are shy on wanting people to buy the software if they can

**Reply by afonsofrancof** [reply]
> We don't want to give the impression that we are shy on wanting people to buy the software if they can

I understand how "Buy" would make more people support Immich, which is really good. The change overall is great, it was just a nitpick.

Love the badge btw, reminds me of the Arc browser

**Reply by dvejsada** [reply]
> We don't want to give the impression that we are shy on wanting people to buy the software if they can

Aren't you concerned about the potential legal implication of "selling" the product with respect to warranties or liability for damages? Under most legal systems (especially in the EU) warranty on the product and liability for damages caused by the product arises directly from the law in case of sale of goods or services and cannot be waived by the seller (even by setting permissive open source licence). From this perspective, supporting the creators (i.e. donation) would seem to be safer than selling a product...

**Reply by alextran1502** [reply]
@dvejsada I am not personally concerned since this is the same pattern many of FUTO's products use. We have a lawyer who helped us take a look at all this. If you know of any lawyers who are concerned about this, please let us know, and we can redirect the questions and concerns to the right party.

**Reply by thanosapollo** [reply]
Using the word "buy" for AGPL-licensed software like Immich, which is free and comes with no warranty, can be misleading or problematic, especially in jurisdictions with strong consumer protection laws, such as the European Union. The term "buy" suggests a commercial transaction, which is not applicable here. 

"Support" or "donate" would be more accurate terms, as they convey the correct nature of the involvement. I've donated to Immich before and would like to do so again.

Is there any intention to change the licensing of Immich from AGPL to a non-free license?

**Reply by alextran1502** [reply]
@thanosapollo there is no intention to change the license of Immich source code. It will always be AGPLv3

**Reply by raffomania** [reply]
The GNU project, and IIRC also Richard Stallman, [explicitly encourage selling free software](https://www.gnu.org/philosophy/selling.html).

**Reply by lightweight** [reply]
As someone whose 'sold free software' for [his entire career](https://davelane.nz/my-open-history), the key is to be precise about *what you're selling*. You're not selling a license or the code. You're selling services related to the code or media for distribution (which is not really relevant these days). Giving people the easy option to "support" the project would be, by far, a better wording than "buying" the software, which, strictly speaking, isn't what's happening. It's misleading.

**Reply by MickLesk** [reply]
@thanosapollo 

> 
"Offering AGPL-licensed software for sale does not in itself violate the license terms or EU law as long as the terms of the license are met. These conditions include in particular

**1.) Accessibility of the source code:** The source code must be made available to the purchaser.
**2.) License information:** The license terms must be clearly communicated.
**3.) Freedom to redistribute:** The buyer has the right to redistribute, modify and make publicly available the software (including the source code).
Under the AGPL, providers can charge fees for the service, support, maintenance or provision of the software. This means that it is legal to charge money for distributing the software as long as the source code is made available and the above conditions are met.

In summary: Offering AGPL 3.0 licensed software for sale does not violate the license or EU law, provided that the license terms are fully complied with."


But I agree with you, buy is always a not so nice paraphrase, it seems binding and obligatory.

I'm more of a fan of “sponsoring” or “supporting”

**Reply by BoBBer446** [reply]
It is true that the term "buying" is firmly defined in Germany. It grants the buyer many rights that the seller grants.
A purchase is a mutual contract that creates an obligation to exchange an item or other object, in particular a right (purchase of a right) or a set of items, for money (otherwise an exchange) (§§ 433ff. BGB).

The purchase contract obliges the seller to hand over the item to the buyer (see, however, commission, conditional transaction), i.e. to grant him possession and to procure ownership of the item or the right sold (§ 433 I BGB). As a result of the principle of abstraction that governs our law, i.e. the separation between the underlying transaction under the law of obligations and the transaction in rem, the purchase alone does not bring about a change in the legal situation; rather, the seller is only obliged to bring this about on the basis of the purchase contract (sale; transfer of ownership, disposal). The purchase contract can also be concluded subject to a condition, e.g. that the seller succeeds in procuring the item sold (reservation of self-supply); see also purchase on trial. In addition, the seller regularly has an obligation to provide information about the legal circumstances relating to the item sold, to hand over the documents serving as proof of the right (§ 444 BGB), to bear the costs of handover, packaging, in particular the costs of shipping the item sold to the place of performance (sale by delivery to destination and the cif, "fob", carriage paid, carriage paid clauses) and, under certain circumstances, to store the item until delivery, etc. If the seller is obliged to manufacture the item itself in addition to procuring the material, this constitutes a contract for work and materials.

**Reply by MickLesk** [reply]
What you mean, however, relates to normal purchase contracts. Here you have to define it a little differently (also by using the word “purchase”). Ultimately, it's really just a spelling issue, buying is always difficult. 1) There must be contractual clarity as to what you are “buying” here and what the “added benefit” is, since there is no added benefit in the direct sense, you can discuss “misleading”, but that in turn is a matter of communication, as described in the “Immich” program. 
I assume you are also from Germany? Partly the land of poets and thinkers, but partly also a lot of legal bullshit, regulations and laws.

**Reply by BoBBer446** [reply]
Whether it is legal nonsense or not:
In Germany, you enter into a legally valid purchase contract by clicking the "Buy" button. 

Amazon, eBay and other retailers in Germany have also come to realize this. This is also the case in the EU. Whether it is good or bad is up for debate. 

I didn't write anything about legal advice. I'm just saying that in the EU the word "purchase" is firmly defined and the buyer has many rights.

**Reply by basdp** [reply]
> @thanosapollo
> In summary: Offering AGPL 3.0 licensed software for sale does not violate the license or EU law, provided that the license terms are fully complied with."

You forgot the most important part: any licensee (basically all of us, as we now all have this source code and are thus licensed to use it under the AGPL) can freely distribute it FOR FREE, even if the original team would only give the source code after paying. This inherently summarized what is the main flaw with this whole plan. You can't sell licenses to open source software. It wouldn't be 'open' anymore.

**Comment by Skywalker8510** [comment]
The positive response to community feedback makes me so much happier to support the project.

**Comment by Enigmind** [comment]
Perfect ! I was going to buy Immich anyway to support this great project, but I am glad you actually care and listen the community. Wish you all the best.

To make a little suggestion, you could add a badge/role on the discord server for those who purchased ~~a license~~ the product !

**Reply by alextran1502** [reply]
This will be a logistic nightmare to implement, so maybe at some points 😄 we need to get back to working on actual features and fixing bugs hehe

**Comment by jordanwalster** [comment]
Based on what I've read in the past, there's supposed to be no "talking home", so how does the per user supporter status work, whilst also being lifetime? e.g. I run immich at home, but then I move to an installation run by a friend instead - how is this all verified?

**Reply by alextran1502** [reply]
Key validation is completely local to the Immich server. When you buy a key, you receive two keys: A short key, and a long activation key. Both are needed to cryptographically validate your purchase. For convenience, you have the option of pasting in only the product key, which is then used to retrieve the activation key from the payment server. No information about this is retained on the payment server.

**Comment by Manic-87** [comment]
I was one of those who criticized the change because of the wording and the possible implications going paywalled (although you say it won't happen. I just don't trust such stuff if I think the actions taken indicate otherwise).
With these changes all my worries are gone. This reflects exactly how I have seen immich and the team before the "license"-stuff.

Only one more thing: Please make it possible to hide the supporter badge via settings.
When I'm financially a bit more stable I will definitely buy a product key, but I hate seeing such stuff.

**Reply by Skywalker8510** [reply]
Yea it would be nice to have the choice to hide the supporter badge probably just a toggle in settings.

**Reply by alextran1502** [reply]
Yes, planning to do that as well

**Comment by lusu007** [comment]
I like this change.

What about a group on the Discord server? Or some extra community channels?

**Reply by sr20kschmitz** [reply]
OH I like that idea, maybe even something to indicate supporter on your discord name etc.

**Reply by Skywalker8510** [reply]
Someone else also asked this and they said 

the question 
“To make a little suggestion, you could add a badge/role on the discord server for those who purchased a license the product !”

Their response 
“This will be a logistic nightmare to implement, so maybe at some points 😄 we need to get back to working on actual features and fixing bugs hehe”

**Reply by lightweight** [reply]
I'd love an option to participate in the community without needing to compromise my principles & privacy by being forced to use Discord (or some other proprietary chat system). Why not Matrix? Github is bad enough.

**Reply by lusu007** [reply]
Nobody forces you to use Discord. However, the Discord server already has a lot of users. So why not make the Discord server better?

**Reply by lightweight** [reply]
Actually @lusu007, if I want to participate in the dynamic discussion related to this excellent app, I *do* need to use Discord - it is, so far as I know, the *only* place such discussion takes place. I [wrote this a few years back](https://davelane.nz/notslack) to explain why this is an unfortunate imposition on the community and undermines its values.

**Reply by bo0tzz** [reply]
> the only place such discussion takes place

Isn't such discussion also taking place on github, right this moment?

Fwiw, I agree that Discord kind of sucks, but the reality is that for most folks it's the most accessible platform.

**Reply by lightweight** [reply]
> > the only place such discussion takes place
> 
> Isn't such discussion also taking place on github, right this moment?

It's also an imposition on community participants who value openness and freedom to be using (proprietary, Microsoft-owned) Github. I run my own Gitlab and Forgejo instances to avoid imposing on any who want to collaborate with me on my projects. 
 
> Fwiw, I agree that Discord kind of sucks, but the reality is that for most folks it's the most accessible platform.

Yes, it does suck. Mostly because it's a proprietary platform that holds your community's data - you have to give them your data so you can log in, to be able to see any of that data. A similarly functional platform like Matrix - which is fully libre/FOSS - allows participation just as easily (there're many millions with Matrix accounts) and is distributed, so you can even use your own Matrix server (I run a couple).

**Reply by bo0tzz** [reply]
Running our own code forge or chat server would take away valuable resources from developing the software itself. This is unfortunately an area where pragmatism wins over ideology, for us.

**Reply by lightweight** [reply]
> Running our own code forge or chat server would take away valuable resources from developing the software itself. This is unfortunately an area where pragmatism wins over ideology, for us.

heh. Here're [53 of the web services](https://tech.oeru.org/updating-oer-foundation-web-services-july-2024) I run. [Here're a bunch (30ish) more](https://nzoss.nz/services). Plus I run another dozen or more of my own. I'm one guy - I run these *all* single handedly. I also do a LOT of development (more than half of my time). Self-hosting is not a significant overhead, nor cost. It just isn't. Look - I really appreciate that you've made your Immich project Copyleft. It's great. But regarding technology choices, although I *get* why you've done it, your tech decisions do colour your community - the decisions you've made so far effectively sell out your community and its data to proprietary vendors for your convenience, and they undermine the values you imply by having made your code libre in the first place.

**Comment by aildaan** [comment]
This is why I already bought the server support! Keep it going guys and don't ever change directions!

**Reply by Enigmind** [reply]
Happy we are, us, who spared 0.01$

**Comment by solankimihir** [comment]
This just made my day!
Thank you for listening to us and I love it.

**Comment by Skywalker8510** [comment]
quick question if I buy a individual purchase of Immich does that apply to my app regardless of what server I’m on or is it server specific?

**Reply by bo0tzz** [reply]
It applies to you and you can use it on any server.

**Reply by Skywalker8510** [reply]
Awesome thanks

**Comment by bernhard4116** [comment]
I don't mind how you call it, I'll support you anyway! You're doing a great job, thanks a lot! ❤️🔥

**Comment by justin-pierce** [comment]
Good change! Any reason you're not offering annual options? Something like $10/year. Seems like that'd help make the funding more sustainable (could keep one time versions for people that prefer that).

**Reply by ghan1t** [reply]
I would also like this option better. A lifetime key that includes all future upgrades is not sustainable for a business that isn't growing indefinitely (and nothing can) and I would like to continuously contribute a smaller amount to immich's development. 
It's probably extra work to implement, so I understand if they don't have it available yet, but please @alextran1502 consider it in the future.

**Reply by alextran1502** [reply]
We also thought about other channels for revenue as well, especially when the core Immich is stable, such as offering hosting services for those who don't want to self-host, selling merch...etc

**Reply by Jurrer** [reply]
@alextran1502 
you could provide a BTC address or buy a coffee link for those who want to support immich even after buying """license"""

**Reply by Jurrer** [reply]
nvm, found the answer
https://github.com/immich-app/immich/discussions/11313#discussioncomment-10131001

**Comment by zatricky** [comment]
How is this better? It still gives the impression that it is not open source.

**Reply by alextran1502** [reply]
If anyone has that impression, maybe you can help us point them to our GitHub repository. I think that would help clear up the confusion

**Reply by zatricky** [reply]
Fine, removing some of the things, like "licensed/unlicensed" is arguably better. But "Buy Immich" and the naming of the "editions" is still a dark pattern. It will still confuse people who think they need two licenses in order to add a second user. It still gives the impression that it is not open source.

**Reply by alextran1502** [reply]
We plan to add more FAQ around those things as well. I don't think explicitly stating Buy Immich is a dark pattern because it shows our explicit intention that you should buy software if you can, but we are not forcing you to.

And if you think Immich looks polish like commercial software, I take that as a compliment 😛 Thank you

**Reply by zatricky** [reply]
> If anyone has that impression, maybe you can help us point them to our GitHub repository. I think that would help clear up the confusion

I realise you have no ill intentions - and you don't have to convince me that Immich is open source - but you really should not rely on us to fix bad communications on the project's behalf.

**Reply by levifig** [reply]
Beware that "open source" does not equate "free". There are a LOT of "Commercial Open Source" projects (it's actually a growing market)! Personally, I don't like how most of them structure their licensing, but believe Immich has done it properly, both in terms of format and in terms of pricing.

**Comment by glaeqen** [comment]
@alextran1502 I don't mean to be petty or anything, I just wondered what about people that already GH-sponsored you for quite some time (I think I donated ~$100 already and it keeps on going). Is there a chance to get the "supporter" status / key that way? I don't plan to cancel my sponsorship as I find it more fitting to continuously support the project :) Cheers

**Reply by alextran1502** [reply]
We are working on a way for past donors to claim the key directly. Probably have that done sometimes soon

**Reply by JW-CH** [reply]
Alex already answered this in another post. Just hit the team up and they will see what they can do.

**Reply by glaeqen** [reply]
No stress, keep up the good work 👌🏻

**Reply by glaeqen** [reply]
> Alex already answered this in another post. Just hit the team up and they will see what they can do.

Sorry for repeating the question then, missed that.

**Reply by JW-CH** [reply]
> > Alex already answered this in another post. Just hit the team up and they will see what they can do.
> 
> Sorry for repeating the question then, missed that.

No worries! It was not in this discussion. A FAQ would probably serve good here :)

**Comment by Blisterexe** [comment]
looks great, but having the option to hide the badge supporter once paid would be nice

**Reply by alextran1502** [reply]
Yes, planning to do that as well :D

**Comment by LaurentAtJoes** [comment]
Good job.
Quick question : will you guys be able to provide some kind of invoice or receipt ?
I myself would need just 2 user level keys, but I'd gladly purchase a server level one through my company if I can get something for the accounting.

**Reply by bo0tzz** [reply]
Send an email to billing@futo.org, they should be able to help you out.

**Comment by IrrerPolterer** [comment]
Thanks Immich team, I love this solution!
I was one of those people who immediately purchased the server license, because I think great open source work needs to be supported. But at the same time, I too felt the headache about the phrasing and presentation of this paid tier... This is one of those moments where I see my trust in humanity, communication and the sense of community restored. Great to see an awesome project team taking feedback from their awesome community ❤️

**Comment by roschaefer** [comment]
Hello :wave:  I contributed code some weeks ago. Am I considered a supporter?
https://github.com/immich-app/immich/graphs/contributors

**Reply by alextran1502** [reply]
At the moment, we are considering sending out a key to past financial donors to the project since we have a clear spreadsheet to work with

**Comment by autumn-wind-dev** [comment]
Thank you for your thought and consideration on this Alex et al, this feels like a very good balance 🥳

**Comment by Mansour-J** [comment]
Folks, this is called "Happy Ending" lol

**Comment by charles-997** [comment]
This is really fantastic. Thank you for listening and considering feedback from the community. Happy to buy Immich.

I agree with avoiding the connotation of "charity".  This wording is a perfect balance. Looking forward to more great releases.

**Comment by Biliskn3r** [comment]
Perfect. Thanks for your hard work!!

**Comment by Kixel** [comment]
Very much appreciate this change.
 The whole license/trial thing felt very guilt-trippy, and I really hate that. This actually put me off supporting Immich as a whole.
This new wording (and UI design) IMO makes it much more clear what your mission is and what your intentions are.

**Comment by eigenphase** [comment]
Flooding this with a another big thank you comment for listening to your community!
This looks like to a good solution to me.

This is the way also I will support via the "server" option once I have it installed and running.

**Comment by hofq** [comment]
Will you be able to "Upgrade" multiple individual licenses to a higher one?

For reference, see the "Unraid" pricing scheme, i really love the politics behind the different versions

**Reply by alextran1502** [reply]
I think we will implement that in the subsequent steps. Right now you can just email us and we can swap out for a server key

**Comment by jantari** [comment]
Very happy with this, but a question: when I purchase a server license, and in the future I may decide to move my licensed immich instance from docker compose on my NAS to a VM on another computer, can I reuse the license or do I have to purchase again? basically, are the licenses/product keys bound to the hardware or instance in any way?

**Reply by alextran1502** [reply]
You can reuse the key, the key doesn't bind to anything

**Comment by dragonhearttheone** [comment]
I have no problems with something that shows up once and can be hidden indefinitely. Really appreciate you for working with the community on this!

**Reply by alextran1502** [reply]
The panel is actually not a popup. It only shows when you hover over the Buy button 😛

**Comment by drizuid** [comment]
Hi, i have no plans to use immich at this time, but I think the way you handled this issue is a great example for others in the OSS community to follow, due to this and the hard-work and effort your team have put into immich, do you accept purely monetary donations, I did not see a gh sponsor link or anything.

**Reply by alextran1502** [reply]
Thank you. We turned off our donation page when we joined FUTO. Once you are ready to use Immich, feel free to buy a product key to support us 😃

**Reply by Jurrer** [reply]
@alextran1502 what was the reason to disable this option?

**Comment by PovilasID** [comment]
Step in right direction but I still see deceptive wording by using "buy". Buy drives more sales than "Support" because people might think they do not have all the features and they do not want to explain to their loved ones why that button is there.
I realize that free software or even web will have some dark/gray pastern to try to get the money it is just disappointing to see a project that was an example of doing a lot of things "the right way" slip up.

**Reply by alextran1502** [reply]
I honestly think it is not the dark/gray pattern if you clearly state the reason and give users the consent choice every step of the way. It might create conversations with your users or family about open-source software sustainability, which could be good for discussions.

**Reply by PovilasID** [reply]
We are free to disagree on what is deceptive or what is a clear sales push vs "ethical stance" which just happens to be profitable.

My family does not know what open source is and most people do not and there is not enough time in ones life to care about every cause. I switched them from google photos and if **I** chose to support you will get money you if **I** choose not to support you will not. They do not care about privacy or sustainability. They care about if the thing works and the price of it. If google photos is free and works they will use it. I bet that is the majority of people. R

If any of the people that use it get confused pay for it even if they do not have to I will have to file charge backs. (English is not their first language and the blanket description is useless )

**Reply by nicomerl** [reply]
Why dont you try to explain to your Family what OpenSource is and why your selfhosted Immich is better then Google Photos? 

Its in your own Hands to explane This. Dont try to get lost in details, just explain why data on Immich is private and that 100$ one time is worth it.

If friends and family still dont care about privacy After my Basic explanations, i let them use their cloudservices. I am fine with that.

**Reply by eclairevoyant** [reply]
> people might think they do not have all the features

Threatening chargebacks over reading comprehension is a bit over-the-top. The purchase page explicitly says it grants no extra features. If you have enough money that you can pay $100 without reading what it's for, just because "buy" is on the screen, then congrats on making it in life :slightly_smiling_face:

**Comment by nicomerl** [comment]
> I thought about it when I went to bed, I thought about it in my dream, I thought about it while doing the dishes.

On the one side i Like that you think about Changes Like this alot, on the other side it should not get out of hand.. i was Reading Lots of opinions on Reddit that were Not really fair. People Not wanting to pay anything but praise Immich to Heaven two days before. Yeah, the wording was not Perfect, but the only thing we can do is trust your words and Immich and the Team is absolutely smashing it. This is the one private Solution for photohosting, so maybe thats why people get so emotional. 

Keep up the work and maybe dont read too much Reddit After Mayor changes :D

**Comment by alm059** [comment]
This is the way. Thanks for your work from a happy supporter!

**Comment by daudo** [comment]
thanks for the modifications, I really like the way this has worked out now!

And so I just purchased and activated a server license, but what I fail to see is how to produce an invoice or similar. How can I receive an invoice?

**Reply by bo0tzz** [reply]
Send an email to billing@futo.org, they'll help you out

**Comment by xy-tech** [comment]
Is there a way to donate via BTC or ETH or some other cryptocurrency? I don't see a way to do this on the payment screen and if you guys could provide your public addresses, that'd be swell ;)

**Reply by alextran1502** [reply]
As far as I know, the FUTO team is also working on different payment methods. I will you update if they come up with ways to accept crypto payment

**Reply by DrekkSama** [reply]
I would also interested in this

**Comment by basdp** [comment]
But you don't 'BUY' Immich.. This is still wrong... Also why need this be on EVERY. SINGLE. PAGE. ??

**Reply by alextran1502** [reply]
Part of your instance/server status. We give you option to hide, so you can do that if you don't want to see it

**Comment by basdp** [comment]
> Last but not least, we will add a fancy `Supporter Badge` to recognize those who purchase the product. If you have any ideas to make it "shinier," let us know!

Yes, remove it. It's distracting. It's a photo's app, get out of our way. I will support you if you don't bug me in my photos app' home screen. This is just annoying me...

**Reply by alextran1502** [reply]
We give the user the option to hide it or to leave it.

**Reply by basdp** [reply]
That would be a perfectly acceptable solution for me! :-)

**Reply by MickLesk** [reply]
@alextran1502 
In conjunction with your video - I have one little UI idea. Maybe you like them too?

=> give the possibility to put the supporter badge on the profile picture of the user, not in the sense of replacing the photo, but a fancy frame - this eliminates it at the bottom left
Here is a short example:
https://onecompiler.com/html/42m6sgjzu (click RUN)

**Example:**
- It is purchased
- Users have the options:
1.) hide all
2) Show badge (default)
3) Fancy frame on profile picture :D

**Reply by OmniBlade** [reply]
But if you don't have something for the "product key" to unlock what are you actually buying other than a number?

**Reply by Nordlicht-13** [reply]
You buy the good feeling, that you are supporting the developers.

**Comment by WoofThatByte** [comment]
I don't mind supporting a product that I use.

But what happens If I buy the server license and I migrate my hardware to a new server? Do i lose the license? I mean...is it possible to move the license to another server?

Thanks

**Reply by bo0tzz** [reply]
Validation is purely local, so you can use it on any instance you think it's appropriate to use it on ;)

**Reply by OmniBlade** [reply]
You aren't buying a license, you have a license to use just by virtue of having a copy under the AGPLv3.

**Comment by shields-lv** [comment]
Tonfotos the same price

**Reply by bo0tzz** [reply]
But not open source

**Comment by monotok** [comment]
I really value immich, it is a fantastic project and currently sponsor immich on a monthly basis. In a little over 4 years I would have paid the same as a server license. 
I think adding a monthly option of say $5 a month for those who wish to make an ongoing contribution would be a good idea :)

**Comment by Clear2437** [comment]
Love the changes and that you listened to the community. Just got my server the lifetime $100 supporter badge. Thanks for your awesome product and continued work, keep it up! <3

**Comment by OmniBlade** [comment]
When you sell something you have to convey it to that person... what exactly is being conveyed to someone when they already have a fully licensed install? Be honest and make it something like "Buy a supporters key", because you aren't buying immich the project or immich the name or immich the software license or immich the software support contract. Is there some legalease that sits behind "buy" that clarifies exactly what good or service is being bought?

**Comment by knaak** [comment]
Any idea when people who have been supporting this project via sponsorship will receive their keys?

**Reply by bo0tzz** [reply]
https://buy.immich.app/claim

**Reply by knaak** [reply]
Thank you.

**Comment by leepfrog-ger** [comment]
Are there any options to receive a key for those who donated through liberapay?

**Reply by bo0tzz** [reply]
Please send us an email at claim@immich.app with some details, proof of payment etc. and we'll try to get you hooked up.

**Comment by paul-jewell** [comment]
I understand the need to raise funds for Open Source projects, and already make contributions to the projects I use here. I will be doing so with Immich at some point in the very near future (as in days). One question this (and other examples of working to make open source projects sustainable): As far as I can tell, Immich is built on many other libraries and open source projects - how much of my $100 will be passed upstream to the teams responsible for those projects?

**Reply by alextran1502** [reply]
FUTO sends out grands to other projects every few months. The most recent one that got sent out since we joined, we picked exiftool since we depends heavily on it.

You can watch the video here https://youtu.be/o4k5W3X5veQ?si=PcYtSB7DrLr8XyFJ

**Reply by paul-jewell** [reply]
The projects in the video are all high level user facing projects, and well done to FUTO for the grants. I am thinking more about the "under the hood" projects that provide the glue to enable developers to deliver those projects. This question first came to mind for me when I was looking at Proxmox. They also (quite understandably) would like users to buy a licence to use their product. But their software layer sits on top of a whole ecosystem, which in codebase terms probably exceeds their codebase significantly. Please don't take this as me having a go at the Proxmox developers: I don't know whether they contribute downwards or not, and may be doing them a disservice. 
The examples of small, unsupported software projects being leveraged by the "non-free" companies for nothing are manifest. The behaviour of these companies in treating "Open source" as "free to use" with no *moral* obligation is to me outrageous. Open source projects using open source code have the opportunity to be different.
I think we should be looking at a funding model which trickles down payments to the contributors at all levels. Your project does well, then you benefit, and the developers of the libraries you use also benefit.
Anyway, I don't expect you to change anything based on my comments, but I do ask that you consider them. 
Thank you for Immich: it is a fantastic product, so you and the other developers deserve to be able to make a living out of it!

**Reply by BoBBer446** [reply]
> FUTO sends out grands to other projects every few months. The most recent one that got sent out since we joined, we picked exiftool since we depends heavily on it.
> 
> You can watch the video here https://youtu.be/o4k5W3X5veQ?si=PcYtSB7DrLr8XyFJ

About what WE are talking Here? Is immich and all functions Always free or is there a bit exklusive content in Future?

If WE Talk about a Donation i did Not understood why in hell wer Talk about licences and other Bad wordings If "Donation" is the perfect wordings.

**Comment by 0xFEEDC0DE64** [comment]
I wont pay for immich as long as the android app keeps logging us out every time there is any network issues. This is the main reason we can't completely switch to immich fully

**Reply by BoBBer446** [reply]
Android Version?
Model? I cant say that i Had the Same issue.

**Reply by MickLesk** [reply]
Never heard of such a Bug. I have been using immich for almost 1 year and it just keeps running and running and running.

**Reply by Nordlicht-13** [reply]
> I wont pay for immich as long as the android app keeps logging us out every time there is any network issues. This is the main reason we can't completely switch to immich fully

Do you use BetBird?

**Comment by bodovix** [comment]
Surely 'supprter key/ support immich' would be the correct terminology - seems a bit one foot out the door to me.. is this foss or not...?
